code/game/objects/structures/salvaging.dm

```diff
@@ -0,0 +1,659 @@
+			shoot_projectile(user, /obj/projectile/beam/scatter)
+		if(8 to 19)
+			src.visible_message("<span class='danger'>You see a bright light from the [src] before the laser reactivates in your face!</span>")
+			shoot_projectile(user, /obj/projectile/beam)
+		if(2 to 7)
+			src.visible_message("<span class='danger'>You see a bright light from the [src] before the laser reactivates in your face!</span>")
+			shoot_projectile(user, /obj/projectile/beam/laser/heavylaser)
+		if(1)
+			src.audible_message("<span class='boldwarning'>You hear something crawling out of the [src]!!</span>")
+			if(prob(99))
+				new /obj/item/clothing/mask/facehugger/toy/(get_turf(src)) //gotcha!
+			else
+				new /obj/item/clothing/mask/facehugger/(get_turf(src)) //yeah
+
+/obj/structure/salvageable/destructive_analyzer/proc/shoot_projectile(mob/living/target, obj/projectile/projectile_to_shoot)
+	var/obj/projectile/projectile_being_shot = new projectile_to_shoot(get_turf(src))
+	projectile_being_shot.preparePixelProjectile(get_step(src, pick(GLOB.alldirs)), get_turf(src))
+	projectile_being_shot.firer = src
+	projectile_being_shot.fire(Get_Angle(src,target))
+
+/obj/structure/salvageable/server
+	name = "broken server"
+	icon_state = "wreck_server"
+	salvageable_parts = list(
+		/obj/item/stack/sheet/glass/two = 80,
+		/obj/item/stack/cable_coil/cut = 80,
+		/obj/item/stack/ore/salvage/scrapuranium/five = 60,
+		/obj/item/stack/ore/salvage/scrapmetal/five = 60,
+		/obj/item/stack/ore/salvage/scrapbluespace = 60,
+
+		/obj/item/research_notes/loot/tiny = 50,
+		/obj/item/research_notes/loot/medium = 20,
+		/obj/item/research_notes/loot/big = 5, //you have a chance at summoning god damn ripley lobster from this thing, might as well
+
+		/obj/item/disk/tech_disk/major = 3,
+		/obj/item/disk/tech_disk = 20,
+		/obj/item/disk/data = 20,
+		/obj/item/disk/holodisk = 20,
+		/obj/item/disk/plantgene = 20,
+
+		/obj/item/computer_hardware/network_card = 40,
+		/obj/item/computer_hardware/network_card = 40,
+		/obj/item/computer_hardware/processor_unit = 40,
+		/obj/item/computer_hardware/processor_unit = 40,
+		/obj/item/stock_parts/subspace/amplifier = 40,
+		/obj/item/stock_parts/subspace/amplifier = 40,
+		/obj/item/stock_parts/subspace/analyzer = 40,
+		/obj/item/stock_parts/subspace/analyzer = 40,
+		/obj/item/stock_parts/subspace/ansible = 40,
+		/obj/item/stock_parts/subspace/ansible = 40,
+		/obj/item/stock_parts/subspace/transmitter = 40,
+		/obj/item/stock_parts/subspace/transmitter = 40,
+		/obj/item/stock_parts/subspace/crystal = 30,
+		/obj/item/stock_parts/subspace/crystal = 30,
+		/obj/item/computer_hardware/network_card/advanced = 20
+	)
+
+/obj/structure/salvageable/server/dismantle(mob/living/user)
+	. = ..()
+	var/danger_level = rand(1,100)
+	switch(danger_level) //ever wanted the extreme danger of turn based rng but in space station 13?
+		if(46 to 100)
+			src.audible_message("<span class='notice'>The [src] makes a crashing sound as its salvaged.</span>")
```

Comment:
src.proc is redundant, also the [src] is redundant

Reply:
on all of these yeah

Reply:
src. is removed